File: 14-08/multiMatrizComp.py


# Entrada do número de linhas e de colunas das matrizes
nLinhas = int(input("Digite o numero de linhas da matriz 1: "))
nColunas1 = int(input("Digite o numero de colunas da matriz 1/linhas da matriz 2: "))
nColunas2 = int(input("Digite o numero de colunas da matriz 2: "))
# Declaração das matrizes
matriz1 = []
matriz2 = []
resultado =[]

# Entrada de dados da matriz1
for i in range(nLinhas):
    linha = []
    for j in range(nColunas1):
        valor = float(input(f"Digite o elemento [{i}][{j}] da matriz1: "))
        linha.append(valor)
    matriz1.append(linha)

# Entrada de dados da matriz2
for i in range(nColunas1):
    linha = []
    for j in range(nColunas2):
        valor = float(input(f"Digite o elemento [{i}][{j}] da matriz2: "))
        linha.append(valor)
    matriz2.append(linha)

# Mostrando as matrizes
print("Matriz 1:")
for linha in matriz1:
    print(linha)

print("Matriz 2:")
for linha in matriz2:
    print(linha)


# multiplicação de matrizes
for i in range(nLinhas):
    linha_resultado = []
    for j in range(nColunas2):
        resultado_elemento = 0  
        for k in range(nColunas1):
            resultado_elemento += matriz1[i][k] * matriz2[k][j]
        linha_resultado.append(resultado_elemento)  
    resultado.append(linha_resultado) 

# Exbindo resultado
print("Resultado: ")
for linha in resultado:
    print(linha)

File: 14-08/multiMatrizComp.c

#include <stdio.h>

int main(){

	// Declaração de variaveis
	int nLinhas1, nColunas1, nColunas2, linhaMult, colunaMult;
	
	// Entrada do numero de linhas e de colunas das matrizes
	printf("Digite o numero de linhas da matriz1: ");
	scanf("%d", &nLinhas1);
	printf("Digite o numero de colunas da matriz1(mesmo numero de linhas da matriz2): ");
	scanf("%d", &nColunas1);
	printf("Digite o numero de colunas da matriz2: ");
	scanf("%d", &nColunas2);
	
    // Declaração das matrizes
	double matriz1[nLinhas1][nColunas1];
	double matriz2[nColunas1][nColunas2];
	double resultado[nLinhas1][nColunas2];

    // Entrada de dados das matrizes
   
	 for (int i = 0; i < nLinhas1; i++) {
     	   for (int j = 0; j < nColunas2; j++) {
            resultado[i][j] = 0;
        }
    }

	for(int i = 0; i<nLinhas1; i++){
		for(int j = 0; j<nColunas1;j++){
			printf("Digite o elemento [%d][%d] da matriz1: ", i, j);
			scanf("%lf", &matriz1[i][j]);
		}
	}

	for(int i = 0; i<nColunas1; i++){
		for(int j = 0; j<nColunas2;j++){
			printf("Digite o elemento [%d][%d] da matriz2: ", i, j);
			scanf("%lf", &matriz2[i][j]);
		}
	}


    // Calculo da multiplicação das matrizes
   	for(int i = 0; i<nLinhas1;i++){
		for(int j = 0; j < nColunas2; j++){
			for(int k = 0; k < nColunas1; k++){
				resultado[i][j] += matriz1[i][k] * matriz2[k][j]; 
			}
		}
	}	
    // Exibição do resultado
	printf("Resultado da multiplicação das matrizes:\n");
    for (int i = 0; i < nLinhas1; i++) {
        for (int j = 0; j < nColunas2; j++) {
            printf("%.2lf ", resultado[i][j]);
        }
        printf("\n");
    }

}


